.trellis/scripts/common/session_context.py

```
@@ -91,13 +98,27 @@ def _collect_git_repo_info(name: str, rel_path: str, repo_dir: Path) -> dict | N
-    _, branch_out, _ = run_git(["branch", "--show-current"], cwd=repo_dir)
-    branch = branch_out.strip() or "unknown"
-
-    _, status_out, _ = run_git(["status", "--porcelain"], cwd=repo_dir)
+    status_rc, status_out, _ = run_git(
+        ["status", "--porcelain"],
+        cwd=repo_dir,
+        timeout=_GIT_PROBE_TIMEOUT_SECONDS,
+    )
+    if status_rc != 0:
+        return None
     changes = len([l for l in status_out.splitlines() if l.strip()])
```

Ruff (0.16.0)

[error] Ambiguous variable name: `l`

(E741)